include/koa/authentication/passport-strategies/strategy_services.js

```diff
@@ -0,0 +1,93 @@
+function _getDao(loginContext, pb) {
+    if (loginContext.site) {
+        return new pb.SiteQueryService({
+            site: loginContext.site,
+            onlyThisSite: false
+        });
+    } else {
+        return new pb.DAO();
+    }
+};
+
+function _addUserToSession(req, user, pb) {
+    delete user.password;
+
+    //build out session object
+    user.permissions = pb.PluginService.getPermissionsForRole(user.admin);
+    req.session.authentication.user = user || {};
+    req.session.authentication.user_id = user[pb.DAO.getIdField()] ? user[pb.DAO.getIdField()].toString() : null;
+    req.session.authentication.admin_level = user.admin;
+
+    //set locale if no preference already indicated for the session
+    if (!req.session.locale) {
+        req.session.locale = user.locale;
+    }
+
+    delete req.session._loginContext; // Remove login context from session now its used
+};
+
+function _getQuery(loginContext, pb, ignorePassword) {
+    const usernameSearchExp = pb.regexUtil.getCaseInsensitiveExact(loginContext.username);
```

Comment:
validate that the username is not empty before calling the db , also do you think we need to sanitize the string prior to send to the DB?

Reply:
That's already validated in the Front End with Angular, do you want me to add the same validation in the Back End?

Reply:
If we are doing a "falsy" check somewhere... that should satisfy that.  As for the issue with sanitizing user data.  We need to make sure user typed data is sanitized before going to the DB.  I do not remember if PB does this for this controller or not.  Might be worth checking.